packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -63,14 +63,40 @@ export function getSPOAndGraphRequestIdsFromResponse(headers: { get: (id: string
     return additionalProps;
 }
 
+export interface IFacetCodes {
+    facetCodes?: string[];
+ }
+
+export function parseFacetCodes(response: string): string[] {
+    const stack: string[] = [];
+    let error;
+    try {
+        error = JSON.parse(response).error;
+    }
+    catch(e) {
+        return stack;
+    }
+
+    // eslint-disable-next-line no-null/no-null
+    if(typeof error === "object" && error !== null) {
+        while (error !== undefined) {
+            if (error.code !== undefined) {
+                stack.push(error.code);
```

Comment:
for a stack i usually use unshift, which will put the element first. this could also remove the need for the reverse when walking the codes